starter_code/queries.md

```diff
@@ -4,80 +4,147 @@
 
 ### 1. All the companies whose name match 'Babelgum'. Retrieve only their `name` field.
 
-<!-- Your Code Goes Here -->
+FILTER:{name: "Babelgum"}
+PROJECT:{_id:0, name: 1}
+SORT:
+SKIP:0		LIMIT:0
 
 ### 2. All the companies that have more than 5000 employees. Limit the search to 20 companies and sort them by **number of employees**.
 
-<!-- Your Code Goes Here -->
+
+FILTER: {number_of_employees: {$gt: 9000}}
+PROJECT
+SORT: {number_of_employees: 1}
+SKIP:0		LIMIT:0
+
 
 ### 3. All the companies founded between 2000 and 2005, both years included. Retrieve only the `name` and `founded_year` fields.
 
-<!-- Your Code Goes Here -->
+FILTER: { $and: [{founded_year: {$gte: 2000}},{founded_year: {$lte: 2005}}]}
+PROJECT: {_id:0 ,name: 1,founded_year: 1}
+SORT
+SKIP:0		LIMIT:0
 
 ### 4. All the companies that had a Valuation Amount of more than 100.000.000 and have been founded before 2010. Retrieve only the `name` and `ipo` fields.
 
-<!-- Your Code Goes Here -->
+FILTER:  { $and:[{"ipo.valuation_amount" : {$gt: 100000000}},{founded_year:{$lt: 2010}}]}
+PROJECT: {_id:0,name:1,ipo:1}
+SORT
+SKIP:0		LIMIT:0
 
 ### 5. All the companies that have less than 1000 employees and have been founded before 2005. Order them by the number of employees and limit the search to 10 companies.
-
-<!-- Your Code Goes Here -->
+FILTER: { $and: [{number_of_employees: {$lt: 1000}},{founded_year: {$lt: 2000}}]}
+PROJECT
+SORT: {number_of_employees: 1}
+SKIP:0		LIMIT:0
 
 ### 6. All the companies that don't include the `partners` field.
 
-<!-- Your Code Goes Here -->
+
+FILTER	: {partners: { $size: 0 } }
+PROJECT	:
+SORT	:
+SKIP:0		LIMIT:0
 
 ### 7. All the companies that have a null type of value on the `category_code` field.
 
-<!-- Your Code Goes Here -->
+
+FILTER	: {category_code: { $eq: null } }
+PROJECT	:
+SORT	:
+SKIP:0		LIMIT:0
 
 ### 8. All the companies that have at least 100 employees but less than 1000. Retrieve only the `name` and `number of employees` fields.
 
-<!-- Your Code Goes Here -->
+
+FILTER	:{$and:[{number_of_employees: {$lt: 1000}},{number_of_employees: {$gte: 100}}]}
+PROJECT	:{_id:0 ,name: 1,number_of_employees: 1}
+SORT	:
+SKIP:0		LIMIT:0
 
 ### 9. Order all the companies by their IPO price in a descending order.
 
-<!-- Your Code Goes Here -->
+
+FILTER	:
+PROJECT	:
+SORT	:{"ipo.valuation_amount" : 1}
+SKIP:0		LIMIT:0
 
 ### 10. Retrieve the 10 companies with more employees, order by the `number of employees`
 
-<!-- Your Code Goes Here -->
+FILTER	:{number_of_employees : {$gte : 186000}}
+PROJECT	:
+SORT	:{number_of_employees : 1}
+SKIP:0		LIMIT:0
```

Comment:
This query is not modulable, you could handle it by sorting them in a descending way and limit it to 10 companies.
SORT	:{number_of_employees : 1}
LIMIT: 10